CMakeLists.txt

```diff
@@ -72,41 +72,39 @@ option(CHRONOLOG_ENABLE_DOXYGEN "Enable Doxygen documentation generation." OFF)
 # TODO Should we consider the posibility of a user passing their own fflags through CMAKE_CXX_FLAGS
 #   Encapsulate everything around if(NOT CMAKE_CXX_FLAGS)
 #   Check Hermes for hwo they delete flags automatically
-if(SET_WERROR)
-    add_compile_options(-Wall)
-    add_compile_options(-Wextra)
-    add_compile_options(-Werror)
-else()
-    add_compile_options(-Wall)
-    add_compile_options(-Wextra)
-endif()
+#if(SET_WERROR)
+#    add_compile_options(-Wall)
+#    add_compile_options(-Wextra)
+#    add_compile_options(-Werror)
+#else()
+#    add_compile_options(-Wall)
+#    add_compile_options(-Wextra)
+#endif()
 
 # Change some compilation flags depending on the build type
 if(CMAKE_BUILD_TYPE STREQUAL "Debug")
     message("Compiling ChronoLog in Debug Mode")
 #    add_compile_options(-g3)
     add_compile_options(-O0)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0")
-
-    if (HERMES_USE_ADDRESS_SANITIZER)
-        add_compile_options(-fsanitize=address)
-    endif()
-
-    if (HERMES_USE_THREAD_SANITIZER)
-        add_compile_options(-fsanitize=thread)
-    endif()
-
-    if (HERMES_USE_THREAD_SANITIZER AND HERMES_USE_ADDRESS_SANITIZER)
-        message(FATAL_ERROR "Cannont use -fsanitize=address and -fsanitize=thread "
-                "at the same time")
-    endif()
-endif()
-
-if(CMAKE_BUILD_TYPE STREQUAL "Release")
+    #set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0 -g")
+
+    #if (HERMES_USE_ADDRESS_SANITIZER)
+    #    add_compile_options(-fsanitize=address)
+    #endif()
+
+    #if (HERMES_USE_THREAD_SANITIZER)
+    #    add_compile_options(-fsanitize=thread)
+    #endif()
+
+    #if (HERMES_USE_THREAD_SANITIZER AND HERMES_USE_ADDRESS_SANITIZER)
+    #    message(FATAL_ERROR "Cannont use -fsanitize=address and -fsanitize=thread "
+    #            "at the same time")
+    #endif()
+else()    
     message("Compiling ChronoLog in release Mode")
-    add_compile_options(-O1)
-    add_compile_options(-Werror)
+    add_compile_options(-O3 -lm)
+    add_compile_options(-Wno-switch-unreachable -Wno-return-type)
 endif()
```

Comment:
I agree with all the compile options except for "no-switch-unreachable" & "no-return-type" ;
these should not be silenced but should be treated as errors and fixed

Reply:
With current `-Werror` flag, warnings such as unused parameters like all these flags in all APIs will become errors. We definitely need to suppress that with flag like `-Wunused`. But `-Wno-switch-unreachable` and `-Wno-return-type` should be fixed.

Reply:
@fkengun Yeah, I have kept your previous version with the Debug build and suppressed warnings in the Release build.